server/modules/user/repository/userInvitationRepository.js

```diff
@@ -0,0 +1,56 @@
+import { db } from '@server/db/db'
+
+import * as Survey from '@core/survey/survey'
+import * as User from '@core/user/user'
+import * as camelize from 'camelize'
+import * as DbUtils from '@server/db/dbUtils'
+
+export const tableName = 'user_invitation'
+
+export const tableColumns = ['user_uuid', 'survey_uuid', 'invited_by', 'invited_date', 'removed_date']
+
+export const insertUserInvitation = async ({ user, survey, userToInvite }, client = db) =>
+  client.one(
+    `INSERT INTO ${tableName} (user_uuid, survey_uuid, invited_by)
+      VALUES ($1, $2, $3)
+      RETURNING id`,
+    [User.getUuid(userToInvite), Survey.getUuid(Survey.getSurveyInfo(survey)), User.getUuid(user)]
+  )
+
+export const updateRemovedDate = async ({ survey, userUuidToRemove }, client = db) =>
```

Comment:
it seems quite inefficient to pass survey just to get the uuid.
You can update the user_invitation table with survey_id instead of survey_uuid; this way you won't need to fetch the survey when deleting a user from it.